nilearn/signal.py

```diff
@@ -464,19 +465,38 @@ def clean(signals, runs=None, detrend=True, standardize='zscore',
           ensure_finite=False):
     """Improve :term:`SNR` on masked :term:`fMRI` signals.
 
-    This function can do several things on the input signals, in
-    the following order:
+    This function can do several things on the input signals. With the default
+    options, the procedures are performed in the following order:
 
     - detrend
-    - low- and high-pass filter
+    - low- and high-pass butterworth filter
     - remove confounds
     - standardize
 
     Low-pass filtering improves specificity.
 
     High-pass filtering should be kept small, to keep some sensitivity.
 
-    Filtering is only meaningful on evenly-sampled signals.
+    Butterworth filtering is only meaningful on evenly-sampled signals.
+
+    When performing scrubbing (censoring high-motion volumes) with butterworth
+    filtering, the signal is processed in the following order:
+
+    - interpolate high motion volumes with cubic spline interpolation.
+    - detrend
+    - low- and high-pass butterworth filter
+    - censor high motion volumes
+    - remove confounds
+    - standardize
```

Comment:
Sorry for the noise, but does this order of steps stem from Lindquist2018 (cited below), or is there another paper that should be cited for this section?

Reply:
My understanding is that they do temporal bandpass filtering and nuisance regression simultaneously ("One can define an omnibus projection matrix consisting of all nuisance variables one seeks to remove from the data. This entails performing motion regression, CompCor, temporal filtering, and so on in a common joint model."). I think this gets complicated because, as far as I can tell, they didn't perform any form of censoring, so it isn't totally clear where this should be incorporated.

The approach I am taking is the following, which mirrors Caterina Gratton and Jonathan Power's pipeline, but includes nuisance regression twice so as to avoid confounds influencing the interpolation step, while also performing temporal bandpass filtering and nuisance regression simultaneously:
1. Demean and detrend
2. Nuisance regression
3. Censor volumes with fFD > .1
4. Interpolate over censored TRs
5. Temporal bandpass filtering and nuisance regression
6. Censor volumes with fFD > .1

According to the [NiLearn documentation](https://nilearn.github.io/dev/modules/generated/nilearn.image.clean_img.html), when you specify bandpass filtering in the same call as the nuisance regressors, it performs the method from Lindquist (2019).

Reply:
Thanks for adding the description and reference!